pkg/model/components/etcdmanager/model.go

```diff
@@ -440,6 +441,16 @@ func (b *EtcdManagerBuilder) buildPod(etcdCluster kops.EtcdClusterSpec, instance
 			}
 			config.VolumeNameTag = openstack.TagNameEtcdClusterPrefix + etcdCluster.Name
 
+		case kops.CloudProviderScaleway:
+			config.VolumeProvider = "scaleway"
+
+			config.VolumeTag = []string{
+				fmt.Sprintf("%s=%s", scaleway.TagClusterName, b.Cluster.Name),
+				fmt.Sprintf("%s=%s", scaleway.TagNameEtcdClusterPrefix, etcdCluster.Name),
+				fmt.Sprintf("%s=%s", scaleway.TagNameRolePrefix, scaleway.TagRoleMaster),
+			}
+			config.VolumeNameTag = fmt.Sprintf("%s=%s-%s", scaleway.TagInstanceGroup, instanceGroupName, etcdCluster.Name)
```

Comment:
Why does it have to be IG with Cluster prefix (%s=%s-%s"). There should be no need for the cluster part, as there is already a cluster tag.

Reply:
If the `FindVolumes()` function finds more than 1 volume, the first one of the list gets mounted (either `main` or `events`) but not the other one, therefore the etcd-manager pod which volume is not mounted can't work properly. The only way I managed to prevent this is to make sure that the name tag is unique so that the list returned by `FindVolumes` only contains one element for each pod running the function.

Reply:
Could you check https://github.com/kubernetes-sigs/etcdadm/pull/334? It is an example on how to deal with this issue. This approach combines creates 2 sets of tags:
* matchPeerTags = VolumeTags (used to find all peers of same type)
* matchNameTags = VolumeTags + VolumeNameTag (used to find the exact volume that needs mounting)

Cluster an EtcdClusterPrefix should be enough to match all peers. Adding the InstanceGroup tag should make it unique.

Reply:
I reverted my latest changes on the name tag, now it's only the instance group name so we should be good to go for this one too !

Reply:
Perfect. Thanks!